### PR TITLE
Update checkOrGenerateImageType $sourceFilePath

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Image generation always took the original file (cover), regardless of the type of image
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See #32404
| Fixed ticket?     | #32404
| Related PRs       | -
| Sponsor company   | -

### CODE REVIEW

**Review by boherm — DISMISSED** [review]
LGTM, thanks @M0rgan01!

**Review by mflasquin — DISMISSED** [review]
Seems good, but $originalImagePath variable defined in line 187 is not used anymore ?

**Comment by mflasquin** [comment]
UI tests : https://github.com/mflasquin/ga.tests.ui.pr/actions/runs/5001392233

**Comment by M0rgan01** [comment]
> Seems good, but $originalImagePath variable defined in line 187 is not used anymore ?

@mflasquin you are right, it is no longer used. I removed it

**Review by florine2623 — COMMENTED** [review]
Hello @M0rgan01 ,

Tests autos are green : https://github.com/florine2623/testing_pr/actions/runs/5022232806

I have well installed your PR.
But when I check the category image folder, I don't have the right extension on the files.

I still have the following reported on the issue :
9. The `img/c/{idCategory}/{idCategory}-0_thumb.webp` doesn't exist => ERROR 

Screenshot 2023-05-19 at 11 07 15

Here I should have one more file called `11-0_thumb.webp`

Am I missing something ?
Thanks^^

**Comment by M0rgan01** [comment]
Hi @florine2623 :smile:  Good remark ! 

Currently the generation of wepP images for categories is executed when displaying on FO.

If the FO does not display the image, it will not be generated! And this is the case for the thumbnails.

The issue https://github.com/PrestaShop/PrestaShop/issues/30520 talks about it

>  - Menu thumbnail
>    - Not used in classic theme right now at all.
>    - Not provided in FO $category object at all.

**Comment by Hlavtox** [comment]
Yep, as @M0rgan01 says, there is no way to access the thumbnails in FO until that issue is resolved.

We don't even have a way of defining sizes for these images, unfortunately. image_type table structure needes to be modified.

**Review by boherm — DISMISSED** [review]

**Review by Hlavtox — DISMISSED** [review]
This is not correct fix, it generates a thumbnail from a thumbnail, completely breaking the logic.

It must use the original image to generate it.

Related issue is not caused by ImageRetriever but by some backoffice logic. ImageRetriever is used for FO and it can properly generate all images with a tiny modification.

**Comment by M0rgan01** [comment]
> This is not correct fix, it generates a thumbnail from a thumbnail, completely breaking the logic.
> It must use the original image to generate it.

If the Category cover image is totally different than the Category thumbnail image, we are not going to use the cover image to generate the thumbnail image?

**Comment by Hlavtox** [comment]
@M0rgan01 No, they are a different image, we are currently discussing the best way to structure it.

It should look like this:

- Original image for cover - `/img/c/1.jpg`, working
- Original image for thumbnail - `/img/c/1_thumb.jpg`, working
- Resized image(s) for cover - `/img/c/1-category_default.jpg`, working
- Resized image(s) for thumbnail - `/img/c/1_thumb-category_default.jpg`. Currently broken, backoffice is generating it as `/img/c/3-small_default.jpg` overwriting the cover image miniature.

**Comment by M0rgan01** [comment]
@Hlavtox Yes I understood :smiley: 

The problem comes from the generation of webP files which systematically take the original cover image as a basis.

If my cover image is a :dog: , and my thumbnail is a :cat: , when I go to the FO, the webP generation of my cover will be good (:dog: ), but my thumbnail will also be a :dog: 

:crying_cat_face:

**Comment by Hlavtox** [comment]
> The problem comes from the generation of webP files which systematically take the original cover image as a basis.

This is correct, it must use the original image.

> If my cover image is a 🐶 , and my thumbnail is a 🐱 , when I go to the FO, the webP generation of my cover will be good (🐶 ), but my thumbnail will also be a 🐶

Yes, but the solution is to use original 🐱 (1_thumb.jpg), not a miniature of a 🐶(1-category_default.jpg) ;-)

**Comment by M0rgan01** [comment]
I was fooled... The base dataset does not create a `_thumb.jpg` file, so I did without

image

I pushed new commit @Hlavtox :dog: Thank you for the explanations

**Review by tleon — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Comment by florine2623** [comment]
Hello @M0rgan01 ,

Could you rebase your PR so that the automated tests can pass on it ?
Thanks!

**Comment by M0rgan01** [comment]
Auto tests https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/5403189587

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@M0rgan01 Could you enable UI tests in `UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts` ? It's commented. You can find them with `// @todo : https://github.com/PrestaShop/PrestaShop/issues/32404` (only for the part 9/10 : https://github.com/PrestaShop/PrestaShop/issues/32404#issuecomment-1549745748)

**Comment by M0rgan01** [comment]
Auto tests: https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/5410684680

**Comment by M0rgan01** [comment]
Tests: https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/5412404512

File: tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts [hunk 1]
```diff
@@ -244,17 +244,14 @@ describe('BO - Design - Image Settings - Check category image format', async ()
           const imageTypeCoverJPG = await files.getImageType(pathImageCoverJPG);
           await expect(imageTypeCoverJPG).to.be.eq(arg.extImageType);
 
-          // @todo : https://github.com/PrestaShop/PrestaShop/issues/32404
-          /*
           // Check the WebP file
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/08_design/06_imageSettings/14_checkCategoryImageFormat.ts:247]:
```suggestion
          // @todo : https://github.com/PrestaShop/PrestaShop/issues/32404
          // Check the WebP file
```

**Review by Progi1984 — DISMISSED** [review]

**Review by mflasquin — DISMISSED** [review]

**Review by Hlavtox — COMMENTED** [review]
Are you sure this is going to work properly?

If I understand correctly, when image thumbnail name is `small_default`, it's gonna take `$id_image . '_thumb.jpg'` as a source.

So this will break `small_default` thumbnails for all other objects other than category, because only category has `ID_thumb.jpg` images. It was a hackjob before, but this is not better. All thumbnails MUST contain the same image.

**Review by sallemiines — COMMENTED** [review]
Hello @M0rgan01 
Thanx fo the PR 

[Tests autos](https://github.com/sallemiines/ga.tests.ui.pr/actions/runs/5464559945/jobs/9946757343) OK 

- The img/c/{idCategory}/{idCategory}-category_default.webp doesn't exist => ERROR (Cover)
- The img/c/{idCategory}/{idCategory}-0_thumb.webp doesn't exist => ERROR (Thumb)
- The img/c/{idCategory}/{idCategory}-small_default.webp doesn't exist => ERROR (Menu)

When I go on the front office

- The img/c/{idCategory}/{idCategory}-category_default.webp exist => OK 
- The img/c/{idCategory}/{idCategory}-0_thumb.webp doesn't exist => Nok
- The img/c/{idCategory}/{idCategory}-small_default.webp OK
- img/c/{idCategory}/{idCategory}-small_default.jpg => OK 



**Result of /img/c after creating a new Ctegory** 
image



**Result of img/c after going to FO** 

image

https://github.com/PrestaShop/PrestaShop/assets/31663369/c18a5640-1db3-430b-a360-fae76f4db87b

Please check and feedback ! 
Thank you !

**Comment by M0rgan01** [comment]
@Hlavtox I adjusted the condition for categories. I also think this should be made generic for all image types, but for now I have a quick fix for v8. We will be able to review the way to generate the images of the categories in FO and BO in v9.

@sallemiines As mentioned above, it is not possible to generate the image for the point "The img/c/{idCategory}/{idCategory}-0_thumb.webp doesn't exist => Nok". https://github.com/PrestaShop/PrestaShop/pull/32597#issuecomment-1554588279

**Review by kpodemski — APPROVED** [review]

**Comment by Hlavtox** [comment]
@M0rgan01 This should be good brother, I think it's good enough for v8.1.

We can do it properly for v9. :-) https://github.com/PrestaShop/PrestaShop/issues/30520

Don't worry about the thumbnails, they never worked and not even accessible in FO.

**Review by Hlavtox — APPROVED** [review]

**Comment by Hlavtox** [comment]
@sallemiines Ines, you can retest it. :-) 
Don't worry about `The img/c/{idCategory}/{idCategory}-0_thumb.webp doesn't exist => Nok`

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @M0rgan01 ,

After regenerating the images in Design > Image Settings > Regenerate thumbnails > Categories

Here's what I have :
Screenshot 2023-07-18 at 16 11 26

✅ The img/c/{idCategory}/{idCategory}-category_default.webp doesn't exist => ERROR (Cover)
✅ https://github.com/PrestaShop/PrestaShop/pull/32597#issuecomment-1630830849 The img/c/{idCategory}/{idCategory}-0_thumb.webp doesn't exist => ERROR (Thumb)
✅ The img/c/{idCategory}/{idCategory}-small_default.webp doesn't exist => ERROR (Menu)

✅ The img/c/{idCategory}/{idCategory}-category_default.webp exist => OK (Cover)
✅ https://github.com/PrestaShop/PrestaShop/pull/32597#issuecomment-1630830849 The ✅ img/c/{idCategory}/{idCategory}-0_thumb.webp doesn't exist => ERROR (Thumb)
✅ The img/c/{idCategory}/{idCategory}-small_default.webp exist but it's not the webp version of img/c/{idCategory}/{idCategory}-small_default.jpg => ERROR (Menu)

It is QA ✅ 
Thanks all for the details about the PR

**Comment by mflasquin** [comment]
thanks @M0rgan01

**Pull request merged**